types/src/error.rs

```diff
@@ -183,13 +185,14 @@ impl Error {
 #[derive(Deserialize, Serialize, Debug, PartialEq)]
 pub struct SubscriptionClosedError {
 	subscription_closed: String,
-	id: u64,
+	// TODO: #[serde(borrow)]..
```

Comment:
//cc @maciejhirsz 

I have lifetimes issues with serde, ideally we should have `SubscriptionId<'a>` here but that forces us to change the NotifResponse to 

```
#[derive(Debug, Deserialize, Serialize)]
#[serde(untagged)]
enum NotifResponse<'a, Notif> {
	Ok(Notif),
	#[serde(borrow)]..
	Err(SubscriptionClosedError<'a>),
}
```
which brakes this:

```rust
impl<Notif> Stream for Subscription<Notif>
where
	Notif: DeserializeOwned,
{
	type Item = Result<Notif, Error>;
	fn poll_next(mut self: Pin<&mut Self>, cx: &mut task::Context<'_>) -> task::Poll<Option<Self::Item>> {
		let n: Option<serde_json::Value> = futures_util::ready!(self.notifs_rx.poll_next_unpin(cx));
		let res = n.map(|n| match serde_json::from_value::<NotifResponse<Notif>>(n) {
			Ok(NotifResponse::Ok(parsed)) => Ok(parsed),
			Ok(NotifResponse::Err(e)) => Err(Error::SubscriptionClosed(e)),
			Err(e) => Err(Error::ParseError(e)),
		});
		task::Poll::Ready(res)
	}
}
```

most likely because of `DeserializedOwned` because the clients send `serde_json::Value` back and forth it doesn't work, any ideas?